### ISSUE TITLE
Strange behaviour with wildcard "^"

### ISSUE THREAD

**Issue by gianfrasoft** [keiffster/program-y#189]
We are trying to make this code work:
`    <category>
        <pattern>TEST1 ^ TEST2 ^</pattern>
        <template>answer 2</template>
    </category>`
    
`    <category>
        <pattern>TEST1 ^</pattern>
        <template>answer 1</template>
    </category>`

but:
- when we send the input **TEST1** none of the above pattern matches
- if whe send the input **TEST1 TEST2**, the first pattern works fine
- if whe send the input **TEST1 TESTX**, the second pattern works fine
- (most strange behavoiur) if we remove the first category, the second pattern works fine(!).

* Version of Python: 3
* Operating System ( inc Version ): Ubuntu 16

**Comment by keiffster** [comment]
Thats not very good, let me add a test case and see what is going on

**Comment by keiffster** [comment]
Interesting problem to while away an evening. Found the problem and will be pushing a release to master in the next day or so. 

In the mean time, if its urgent, then you grammar is asking for a match against

TEST1
TEST1 Or Anything else
TEST1 TEST2
TEST1 Anything TEST2
TEST1 Anything TEST2 Anything

This can be replaced with the following patterns

```xml
<pattern>TEST1</pattern>
<pattern>TEST1 *</pattern>
<pattern>TEST1 ^ TEST2 ^</pattern>
```

This will give the same result

**Comment by gianfrasoft** [comment]
Thank you very much @keiffster

**Comment by keiffster** [comment]
This has now been released in 3.6.2
